src/figuresClasses.ts

```diff
@@ -1,11 +1,81 @@
-export interface Figure {}
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: string;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All sides must be greater than 0');
+    }
+
+    const sides = [a, b, c].sort((x, y) => x - y);
+
+    if (sides[2] >= sides[0] + sides[1]) {
+      throw new Error(
+        'The longest side must be less than the sum of the other two sides',
+      );
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
+}
 
-export class Circle implements Figure {}
+export class Circle implements Figure {
+  shape: Shape = 'circle';
 
-export class Rectangle implements Figure {}
+  constructor(
+    public color: Color,
+    private radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Radius must be greater than 0');
```

Comment:
[CODE STYLE] - When throwing errors, provide a detailed description in the error message. Instead of 'Radius must be greater than 0', you could say 'Circle radius must be greater than 0'.